python/templates/Collection.h.jinja2

```diff
@@ -39,8 +45,22 @@ class {{ class.bare_type }}Collection : public podio::CollectionBase {
 public:
   using const_iterator = {{ class.bare_type }}CollectionIterator;
   using iterator = {{ class.bare_type }}MutableCollectionIterator;
+  using const_reverse_iterator = {{ class.bare_type }}CollectionReverseIterator;
+  using reverse_iterator = {{ class.bare_type }}MutableCollectionReverseIterator;
 
   {{ class.bare_type }}Collection();
+  #if __cpp_lib_ranges
+  {{ class.bare_type }}Collection(std::ranges::input_range auto&& range)
+  requires(std::convertible_to<std::ranges::range_value_t<decltype(range)>, {{ class.bare_type }}>)
+  : m_isValid(false), m_isPrepared(false), m_isSubsetColl(false), m_collectionID(0), m_storage() {
+    if constexpr (std::ranges::sized_range<decltype(range)>) {
+      reserve(std::ranges::size(range));
```

Comment:
Just for my understanding, this would call `Collection::reserve` in practice, right? Because that does not exist, unless it is introduced in this PR.

Reply:
Yes, I was reading through the full changeset to determine if I have tests for all new functionality, and this code branch doesn't have a test and (as you point out) will likely fail when passed a `sized_range`. My suggestion would be to simply remove the reserve and just let the usual push_back do that when needed.